packages/components/src/overlay/src/useRestoreFocus.ts

```diff
@@ -26,36 +30,43 @@ export function useRestoreFocus(scope: DomScope, { isDisabled }: UseRestoreFocus
+                const walker = createFocusableTreeWalker(document.body, { accept: isTabbable });
                 walker.currentNode = focusedElement as Node;
 
                 const next = () => {
                     return event.shiftKey ? walker.previousNode() : walker.nextNode() as HTMLElement;
                 };
 
-                // Find the next tabbable element after the currently focused element.
+                // Try to find the next logical element to focus after the currently focused element.
                 let nextElement = next() as HTMLElement;
 
-                // If there is no next element, or it is outside the current scope, move focus to the
-                // next element after the node to restore to instead.
-                if (isNil(nextElement) || !scope.isInScope(nextElement)) {
+                // If the next logical element to focus is in the scope, we don't have to do anything, we can let the browser do his default tab behavior.
+                if (!focusScope.isInScope(nextElement)) {
                     const elementToRestore = elementToRestoreRef.current;
 
                     if (document.body.contains(elementToRestore)) {
+                        // If we haven't found an element or the element we found is not in the overlay scope, it might be a use case for some custom restore logic.
+                        // Try to find a tabbable element next to the element to restore.
                         walker.currentNode = elementToRestore;
 
                         // Skip over elements within the scope, in case the scope immediately follows the node to restore.
                         do {
                             nextElement = next() as HTMLElement;
-                        } while (nextElement === event.currentTarget || scope.isInScope(nextElement));
+                        } while (nextElement === event.currentTarget || focusScope.isInScope(nextElement));
 
+                        // Must prevent the browser from doing is thing since we will takeover and handle the tab ourself.
                         event.preventDefault();
 
                         if (!isNil(nextElement)) {
+                            // If we found a tabbable element, focus the element and exit.
+                            // This is a scenario were we restored the tab order. Instead of focusing the first element available under the overlay, we focus the element next to the overlay trigger.
                             nextElement.focus();
                         } else {
+                            // If we can't find any tabbable element to focus, restore the focus on the overlay trigger element.
+                            // This is also a scenario were we restored the tab order. Instead of focusing the first element available under the overlay, we made sure to focus back to the overlay trigger.
```

Comment:
small type: Where

Reply:
> Just a tiny typo, other than that it makes it clearer so can't complain! I'll do some quick QA today of the different focus cases we were covering to make sure it's all gucci

Perf, let me know if you find something.